File: Model/Assistido.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel;

namespace Model;

[Table("assistidos")]
public class Assistido
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long Id { get; set; }

    [Required]
    [MaxLength(100)]
    public string Nome { get; set;}

    [Required]
    [MaxLength(15)]
    public String Cpf { get; set; }

    [Required]
    [Column(TypeName = "decimal(10,2)")]
    public decimal Saldo { get; set; } = 0;

    [Required]
    public long ConferenciaId { get; set; }

    [ForeignKey(nameof(ConferenciaId))]
    public Conferencia Conferencia { get; set; }
}

File: src/main/java/com/lucas/ssvp_api/model/entities/Assistido.java
package com.lucas.ssvp_api.model.entities;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@EqualsAndHashCode(of = "id")
@Table(name = "assistidos")
public class Assistido {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String nome;

    @Column(nullable = false, unique = true, length = 15)
    private String cpf;

    @Column(nullable = false)
    private String pin;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal saldo = BigDecimal.ZERO;

    @ManyToOne
    @JoinColumn(name = "conferencia_id", nullable = false)
    private Conferencia conferencia;
}
